### PR TITLE
[CPU] Support Qwen3.8 text+video: adding torchcodec, ffmpeg and removing pin_memory

### PR DESCRIPTION
**Pull request by yanbing-j**
## Motivation

This PR is to support Qwen3.8 text+video, by adding torchcodec, ffmpeg and removing pin_memory.
Current, torch is 2.12, torchcodec uses 0.12.0 accordingly. ffmpeg needs <9. Default ffmpeg version in ubuntu 26.04 of xeon.dockerfile is 8.0.1, can meet the requirement.

## Reproduce
Launch server:
`sglang serve --model-path Qwen/Qwen3.8-27B  --device cpu  --disable-overlap-schedule  --tp 6  --reasoning-parser qwen3  --tool-call-parser qwen3_coder --mem-fraction-static 0.8`

Client:
```
from openai import OpenAI
client = OpenAI(
    base_url="http://localhost:30000/v1",
    api_key="EMPTY"
)
response = client.chat.completions.create(
    model="Qwen/Qwen3.8-27B",
    messages=[
        {
            "role": "user",
            "content": [
                {
                    "type": "video_url",
                    "video_url": {
                        "url": "https://qianwen-res.oss-accelerate.aliyuncs.com/Qwen3.5/demo/video/N1cdUjctpG8.mp4"
                    }
                },
                {
                    "type": "text",
                    "text": "Describe what happens in this video."
                }
            ]
        }
    ],
    max_tokens=2048,
    stream=True
)
thinking_started = False
has_thinking = False
has_answer = False
for chunk in response:
    if chunk.choices and len(chunk.choices) > 0:
        delta = chunk.choices[0].delta
        if hasattr(delta, 'reasoning_content') and delta.reasoning_content:
            if not thinking_started:
                print("=============== Thinking =================", flush=True)
                thinking_started = True
            has_thinking = True
            print(delta.reasoning_content, end="", flush=True)
        if delta.content:
            if has_thinking and not has_answer:
                print("\n=============== Content =================", flush=True)
                has_answer = True
            print(delta.content, end="", flush=True)
print()

```
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #34189963274](https://github.com/sgl-project/sglang/actions/runs/34189963274)
Latest PR Test (Extra): :x: [Run #34191317185](https://github.com/sgl-project/sglang/actions/runs/34191317185)
Latest PR Test (AMD ROCm 7.2): :x: [Run #34189963237](https://github.com/sgl-project/sglang/actions/runs/34189963237)

### CODE REVIEW

File: docker/xeon.Dockerfile [hunk 1]
```diff
@@ -8,6 +8,7 @@ RUN apt-get update && \
     apt-get full-upgrade -y && \
     DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
     ca-certificates \
+    ffmpeg \
```
Comment (jianan-gu) [inline docker/xeon.Dockerfile:11]: it is by default can meet ffmpeg needs < 9?

Comment (yanbing-j) [reply docker/xeon.Dockerfile:11]: default version of ffmpeg with ubuntu 24.04 is 6.1.1. Verified version 6.1.1 can meet the requirement.

Comment (jianan-gu) [reply docker/xeon.Dockerfile:11]: do we need to ping the version here in case any update breaks this?

Comment (yanbing-j) [reply docker/xeon.Dockerfile:11]: Sorry for the mistake above, ubuntu 26.04 can match with ffmpeg 8.0.1. I have pinned to 8.0.1.

```
-    ffmpeg \
+    ffmpeg=7:8.0.1-3ubuntu2 \
```

File: python/sglang/srt/utils/video_decoder.py [hunk 2]
```diff
@@ -177,6 +181,8 @@ def _decode_chunk(chunk):
                 results[idx] = future.result()
 
         output = torch.cat(results, dim=0)
+        if self._device == "cpu":
+            return output
         return output if output.is_cuda else output.pin_memory()
```
Comment (mingfeima) [inline python/sglang/srt/utils/video_decoder.py]: skipping pin_memory when self._device == "cpu" changes CUDA. The constructor default is "cpu", and CUDA callers such as mimo_v2.py pass device="cpu" then call get_frames_as_tensor. That path still needs pinned host memory for H2D. Gate with is_cpu() (CPU engine), matching qwen_vl.py:269-270, not decode-device == cpu.

**Review by mingfeima — CHANGES_REQUESTED** [review]

File: docker/xeon.Dockerfile (hunk 1 above)
Comment (mingfeima) [reply docker/xeon.Dockerfile:11]: so unpin ffmpeg version works? i saw cuda did not specify version for it.

File: python/sglang/srt/utils/video_decoder.py (hunk 2 above)
Comment (yanbing-j) [reply python/sglang/srt/utils/video_decoder.py]: Done.

File: docker/xeon.Dockerfile (hunk 1 above)
Comment (yanbing-j) [reply docker/xeon.Dockerfile:11]: Yes, unpinned ffmpeg should be fine here. But, I'm not sure if any update will break this.

**Review by mingfeima — APPROVED** [review]

**Comment by mingfeima** [comment]
@yanbing-j update cookbook.
